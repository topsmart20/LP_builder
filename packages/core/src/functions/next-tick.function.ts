const isSafari =
  typeof window !== 'undefined' &&
  /^((?!chrome|android).)*safari/i.test(window.navigator.userAgent);

const isClient = typeof window !== 'undefined';

// TODO: move to shared
export function nextTick(fn: () => void) {
  // TODO: should this be setImmediate instead? Forgot if that is micro or macro task
  // TODO: detect specifically if is server
  // if (typeof process !== 'undefined' && process.nextTick) {
  //   console.log('process.nextTick?');
  //   process.nextTick(fn);
  //   return;
  // }
  // FIXME: fix the real safari issue of this randomly not working
  if (isSafari || typeof MutationObserver === 'undefined') {
    setTimeout(fn);
    return;
  }
  let called = 0;
  const observer = new MutationObserver(() => fn());
  const element = document.createTextNode('');
  observer.observe(element, {
    characterData: true,
  });
  // tslint:disable-next-line
  element.data = String((called = ++called));
}
