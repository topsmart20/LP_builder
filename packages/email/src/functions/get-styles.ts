export function getStyles(block: any) {
  return block && block.responsiveStyles && block.responsiveStyles.large
}
