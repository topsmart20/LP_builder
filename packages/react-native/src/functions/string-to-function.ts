const fnCache: { [key: string]: Function } = {};

export function stringToFunction(
  str: string,
  expression = true,
  errors?: Error[],
  logs?: string[]
) {
  if (!str || !str.trim()) {
    return () => undefined;
  }

  const cacheKey = str + ':' + expression;
  if (fnCache[cacheKey]) {
    return fnCache[cacheKey];
  }

  // FIXME: gross hack
  const useReturn =
    (expression &&
      !(str.includes(';') || str.includes(' return ') || str.trim().startsWith('return '))) ||
    str.trim().startsWith('builder.run');
  let fn: Function = () => {
    /* intentionally empty */
  };

  try {
    // tslint:disable-next-line:no-function-constructor-with-string-args
    // TODO: use strict and eval
    fn = new Function(
      'state',
      'event',
      'block',
      'builder',
      'Device',
      'update',
      // TODO: block reference...
      // TODO: or just remove with (rootState) in general
      `
          var rootState = state;
          if (typeof Proxy !== 'undefined') {
            rootState = new Proxy(rootState, {
              set: function () {
                return false;
              }
            });
          }
          with (rootState) {
            ${useReturn ? `return (${str});` : str};
          }
        `
    );
  } catch (error) {
    if (errors) {
      errors.push(error);
    }
    const message = error && error.message;
    if (message && typeof message === 'string') {
      if (logs && logs.indexOf(message) === -1) {
        logs.push(message);
      }
    }

    console.warn(`Function compile error in ${str}`, error);
  }

  const final = (fnCache[cacheKey] = (...args: any[]) => {
    try {
      return fn(...args);
    } catch (error) {
      console.warn('Builder custom code error:', error);
      if (errors) {
        errors.push(error);
      }
    }
  });

  return final;
}
