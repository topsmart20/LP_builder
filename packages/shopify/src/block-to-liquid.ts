import { BuilderElement, BuilderContent } from '@builder.io/sdk';
import reduce from 'lodash-es/reduce';
import kebabCase from 'lodash-es/kebabCase';
import size from 'lodash-es/size';

export function modelToLiquid(content: BuilderContent, modelName: string, options: Options = {}) {
  const blocks = content.data && content.data.blocks;

  const { html, css } = regexParse(
    `<div
      class="builder-content"
      builder-content-id=${content.id}
      data-builder-content-id=${content.id}
      data-builder-component=${modelName}
      builder-model=${modelName}
    >
      ${blocks ? blocks.map((block: BuilderElement) => blockToLiquid).join('') : ''}
    </div>`.replace(/\s+/, ' ')
  );

  if (options.extractCss) {
    return { html, css };
  }

  return { html: `<style type="text/css" class="builder-styles">${css}</style>` + html };
}

// TODO: move to core
export type Size = 'large' | 'medium' | 'small' | 'xsmall';
export const sizeNames: Size[] = ['xsmall', 'small', 'medium', 'large'];
const sizes = {
  xsmall: {
    min: 0,
    default: 0,
    max: 0,
  },
  small: {
    min: 320,
    default: 321,
    max: 640,
  },
  medium: {
    min: 641,
    default: 642,
    max: 991,
  },
  large: {
    min: 990,
    default: 991,
    max: 1200,
  },
  getWidthForSize(size: Size) {
    return this[size].default;
  },
  getSizeForWidth(width: number) {
    for (const size of sizeNames) {
      const value = this[size];
      if (width <= value.max) {
        return size;
      }
    }
    return 'large';
  },
};

export interface Options {
  emailMode?: boolean;
  extractCss?: boolean;
}

export function blockToLiquid(block: BuilderElement, options: Options = {}): string {
  const css = blockCss(block, options);

  // TODO: utilities for all of this
  const attributes = mapToAttributes({
    ...block.properties,
    ['builder-id']: block.id,
    class: `builder-block ${block.id}${block.class ? ` ${block.class}` : ''}`,
    // TODO: style bindings and start animation styles
  });

  const tag = block.tagName || ((attributes as any).href ? 'a' : 'div');

  // Fragment? hm
  return `
    ${css.trim() ? `<style>${css}</style>` : ''}
    ${
      block.repeat
        ? `{% for ${block.repeat.collection || block.repeat.collection + 'Item'} in ${
            block.repeat.collection
          } %}`
        : ''
    }
    <${tag}${attributes ? ' ' + attributes : ''}>
      ${
        block.children
          ? block.children.map((child: BuilderElement) => blockToLiquid(child, options)).join('')
          : ''
      }
    </${tag}>
    ${block.repeat ? '{% endfor %}' : ''}
    `;
}

const regexParse = (html: string) => {
  const cssSet = new Set();
  const newHtml = html.replace(/<style.*?>([\s\S]*?)<\/style>/g, (match, cssString) => {
    cssSet.add(cssString);
    return '';
  });
  return {
    css: Array.from(cssSet.values())
      .join(' ')
      .replace(/&gt;/g, '>')
      .replace(/&amp;/g, '&')
      .replace(/&quot;/g, '"')
      .replace(/&apos;/g, "'")
      .replace(/ \S+:\s+;/g, '')
      .replace(/\s+/g, ' ')
      .trim(),
    html: newHtml,
  };
};

interface StringMap {
  [key: string]: string | undefined | null;
}

function mapToAttributes(map: StringMap) {
  if (!size(map)) {
    return '';
  }
  return reduce(
    map,
    (memo, value, key) => {
      return memo + ` ${key}="${value}"`;
    },
    ''
  );
}

function mapToCss(map: StringMap, spaces = 2, important = false) {
  return reduce(
    map,
    (memo, value, key) => {
      return (
        memo +
        (value && value.trim()
          ? `\n${' '.repeat(spaces)}${kebabCase(key)}: ${value + (important ? ' !important' : '')};`
          : '')
      );
    },
    ''
  );
}

// TODO: make these core functions and share with react, vue, etc
// TODO: apply style bindings and default animation
function blockCss(block: BuilderElement, options: Options = {}) {
  // TODO: handle style bindings
  const self = block;

  const baseStyles: Partial<CSSStyleDeclaration> = {
    ...(self.responsiveStyles && self.responsiveStyles.large),
  };

  let css = options.emailMode
    ? ''
    : `.builder-block.${self.id} {${mapToCss(baseStyles as StringMap)}}`;

  const reversedNames = sizeNames.slice().reverse();
  if (self.responsiveStyles) {
    for (const size of reversedNames) {
      if (options.emailMode && size === 'large') {
        continue;
      }
      if (
        size !== 'large' &&
        size !== 'xsmall' &&
        self.responsiveStyles[size] &&
        Object.keys(self.responsiveStyles[size] as any).length
      ) {
        // TODO: this will not work as expected for a couple things that are handled specially,
        // e.g. width
        css += `\n@media only screen and (max-width: ${sizes[size].max}px) { \n${
          options.emailMode ? '.' : '.builder-block.'
        }${self.id + (options.emailMode ? '-subject' : '')} {${mapToCss(
          self.responsiveStyles[size] as any,
          4,
          options.emailMode
        )} } }`;
      }
    }
  }
  return css;
}
