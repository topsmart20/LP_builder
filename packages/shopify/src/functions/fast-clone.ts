export const fastClone = <T extends object>(object: T): T => JSON.parse(JSON.stringify(object));
