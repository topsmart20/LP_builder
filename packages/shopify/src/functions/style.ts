import { mapToCss } from './map-to-css';
import { StringMap } from '../interfaces/string-map';

export function style(map: Partial<CSSStyleDeclaration>) {
  return mapToCss(map as StringMap, 0);
}
