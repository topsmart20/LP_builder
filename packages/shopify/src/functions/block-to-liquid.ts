import { BuilderElement } from '@builder.io/sdk';
import { size, set, last, uniq, identity } from 'lodash';
import { sizes, sizeNames } from '../constants/sizes';
import { Options } from '../interfaces/options';
import { StringMap } from '../interfaces/string-map';
import { mapToCss } from './map-to-css';
import { fastClone } from './fast-clone';
import { getComponentInfo } from '../constants/components';

import '../components/text';
import '../components/columns';
import '../components/image';
import '../components/assign';
import '../components/fragment';
import '../components/capture';
import '../components/section';
import '../components/button';
import '../components/symbol';
import '../components/condition';
import '../components/unless';
import '../components/paginate';
import '../components/for';
import '../components/custom-code';
import '../components/state-provider';
import '../components/forms/form';
import '../components/forms/label';
import '../components/forms/input';
import '../components/forms/submit-button';
import '../components/widgets/accordion';
import '../components/widgets/carousel';
import '../components/shopify-section';
import '../components/theme-provider';
import '../components/raw-img';

import { isValidLiquidBinding } from './is-valid-liquid-binding';
import { mapToAttributes } from './map-to-attributes';

const voidElements = new Set([
  'area',
  'base',
  'br',
  'col',
  'embed',
  'hr',
  'img',
  'input',
  'link',
  'meta',
  'param',
  'source',
  'track',
  'wbr',
]);

const isLiquidRender = (binding: string) => {
  return binding.replace(/\s/g, '').includes('shopify.liquid.render');
};

const isLiquidConditional = (binding: string) => {
  return binding.replace(/\s/g, '').includes('shopify.liquid.condition');
};

const getConditionalAttr = (value: string, noEnd = false): string => {
  const closingTag = noEnd ? '' : '{% endif %}';
  return (
    value
      .split('/*start*/')
      .reverse()
      .filter(st => st.includes('shopify'))
      .map(statement => statement.replace(/`/g, '"'))
      .map(statement => {
        const expression = statement.startsWith('!')
          ? 'else'
          : !statement.includes('!')
          ? 'if'
          : '';
        const condition =
          expression === 'if'
            ? statement.match(/context\.shopify\.liquid\.condition\("([^"]*)"/)?.[1]
            : '';
        const index = statement.indexOf('&&');
        const branchValue =
          index > -1 ? getConditionalAttr(statement.substr(index + 2), true) : getValue(statement);

        if (expression) {
          return `{% ${expression} ${condition} %} ${branchValue}`;
        }
        return branchValue + '{% endif %}';
      })
      .join('') + closingTag
  );
};

const removeShopifyContext = (str: string) => {
  return str.replace(
    /(context|state)\s*\.\s*shopify\s*\.\s*liquid\s*\.\s*(get|render)\s*\(\s*(\\"|")([^"]+)(\\"|")\s*.*\)/g,
    '$4'
  );
};

const getValue = (condition: string) => {
  const value = condition.match(/\? (.*) :/)?.[1];
  if (value) {
    return removeShopifyContext(value.replace(/{{'(.*?)'}}/g, '$1').replace(/'/g, ''));
  }
};

/**
 * Extract a liquid expression from our JS structure - i.e. transform
 * "context.shopify.liquid.condition('some | liquid')" to "some | liquid"
 */
export const getLiquidConditionExpresion = (expression: string) => {
  const matched = expression.match(
    /context\s*\.\s*shopify\s*\.\s*liquid\s*\.\s*condition\s*\(\s*['"]([\s\S]*?)['"]\s*,\s*state\s*\)\s*/i
  )?.[1];

  return matched || 'null';
};

// TODO: move most or all of this to transformers and functions
const convertBinding = (binding: string, options: Options) => {
  let value = binding;
  const isShopifyContext = value.replace(/\s/g, '').includes('shopify.liquid');

  if (!isValidLiquidBinding(binding)) {
    return '';
  }

  if (isLiquidConditional(value)) {
    value = getConditionalAttr(value);
  } else if (isShopifyContext) {
    value = removeShopifyContext(value);
  }

  // We use state, Shopify uses global vars, so convert
  // state.product.title to {{ product.title}}, etc
  if (value.includes('state.')) {
    value = value.replace(/state\./g, '');
  }

  if (value.includes('context.')) {
    value = value.replace(/context\./g, '');
  }

  return value;
};

export function blockToLiquid(json: BuilderElement, options: Options = {}): string {
  const block = fastClone(json);

  const styles: StringMap = {};

  const bindings = {
    ...block.bindings,
    ...(block as any).code?.bindings,
  };

  if (bindings) {
    for (const key in bindings) {
      const binding = bindings[key];
      if (!key || !binding || key === 'hide') {
        continue;
      }

      const value = convertBinding(binding, options);

      let valueString;
      if (isLiquidRender(binding) || isLiquidConditional(binding)) {
        valueString = value;
      } else {
        valueString = `{{ ${value} }}`;
      }

      if (key.startsWith('properties.') || !key.includes('.')) {
        if (!block.properties) {
          block.properties = {};
        }
        const name = key.startsWith('properties.') ? key.replace(/^\s*properties\s*\./, '') : key;
        set(block.properties, name, valueString);
      } else if (key.startsWith('component.options.') || key.startsWith('options.')) {
        const name = key.replace(/^.*?options\./, '');
        if (!block.component) {
          continue;
        }
        if (!block.component.options) {
          block.component.options = {};
        }
        set(block.component.options, name, valueString);
      } else if (key.startsWith('style.')) {
        const name = key.replace('style.', '');
        set(styles, name, valueString);
      } else if (key === 'attr.style') {
        if (!block.properties) {
          block.properties = {};
        }
        set(block.properties, 'style', valueString);
      }
    }
  }

  // TODO: bindings with {{}} as values
  const css = blockCss(block, options);

  const stylesList: string[] = [];
  if (size(styles)) {
    stylesList.push(mapToCss(styles, 0));
  }
  if (block.properties?.style) {
    stylesList.push(block.properties.style);
  }

  const bindingClass = block.bindings?.class && convertBinding(block.bindings.class, options);
  const classes = uniq([
    'builder-block',
    block.id,
    block.class,
    bindingClass,
    block.properties?.class,
  ]).filter(identity);
  // TODO: utilities for all of this
  const attributes = mapToAttributes({
    ...block.properties,
    ['builder-id']: block.id,
    class: classes.join(' '),
    ...(size(stylesList) && {
      style: stylesList.join(';'),
    }),
    // TODO: style bindings and start animation styles
  });

  const tag = block.tagName || (block.properties && block.properties.href ? 'a' : 'div');

  const componentInfo = block.component && getComponentInfo(block.component.name);

  if (block.component && !componentInfo) {
    console.warn(`Could not find component: ${block.component.name}`);
  }

  let collectionName =
    block.repeat &&
    last(
      (block.repeat.collection || '')
        .trim()
        .split('(')[0]
        .trim()
        .split('.')
    );

  if (collectionName) {
    collectionName = convertBinding(collectionName, options);
  }

  const componentStr =
    componentInfo && componentInfo.noWrap
      ? componentInfo.component(block, options, attributes)
      : `
      <${tag}${attributes ? ' ' + attributes : ''}>
        ${(componentInfo && componentInfo.component(block, options)) ||
          (block.children &&
            block.children
              .map((child: BuilderElement) => blockToLiquid(child, options))
              .join('\n')) ||
          ''}
      ${!voidElements.has(tag) ? `</${tag}>` : ''}`;

  if (options.componentOnly) {
    return componentStr;
  }

  return `
    ${css.trim() ? `<style>${css}</style>` : ''}
    ${
      block.repeat && block.repeat.collection
        ? `{% for ${block.repeat.itemName || collectionName + '_item'} in ${convertBinding(
            block.repeat.collection,
            options
          )} %}`
        : ''
    }
    ${
      bindings.hide
        ? `{% unless  ${
            !isValidLiquidBinding(bindings.hide) ? 'false' : convertBinding(bindings.hide, options)
          } %}`
        : ''
    }
    ${
      bindings.show
        ? `{% if  ${
            !isValidLiquidBinding(bindings.show) ? 'false' : convertBinding(bindings.show, options)
          } %}`
        : ''
    }
    ${componentStr}
    ${bindings.hide ? '{% endunless %}' : ''}
    ${bindings.show ? '{% endif %}' : ''}
    ${block.repeat && block.repeat.collection ? '{% endfor %}' : ''}
  `;
}

// TODO: make these core functions and share with react, vue, etc
// TODO: apply style bindings and default animation
function blockCss(block: BuilderElement, options: Options = {}) {
  // TODO: handle style bindings
  const self = block;

  const baseStyles: Partial<CSSStyleDeclaration> = {
    ...(self.responsiveStyles && self.responsiveStyles.large),
  };

  let css = options.emailMode
    ? ''
    : `.builder-block.${self.id} {${mapToCss(baseStyles as StringMap)}}`;

  const reversedNames = sizeNames.slice().reverse();
  if (self.responsiveStyles) {
    for (const size of reversedNames) {
      if (options.emailMode && size === 'large') {
        continue;
      }
      if (
        size !== 'large' &&
        size !== 'xsmall' &&
        self.responsiveStyles[size] &&
        Object.keys(self.responsiveStyles[size] as any).length
      ) {
        // TODO: this will not work as expected for a couple things that are handled specially,
        // e.g. width
        css += `\n@media only screen and (max-width: ${sizes[size].max}px) { \n${
          options.emailMode ? '.' : '.builder-block.'
        }${self.id + (options.emailMode ? '-subject' : '')} {${mapToCss(
          self.responsiveStyles[size] as any,
          4,
          options.emailMode
        )} } }`;
      }
    }
  }
  return css;
}
