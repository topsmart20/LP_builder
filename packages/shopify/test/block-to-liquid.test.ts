import { blockToLiquid } from '../src/functions/block-to-liquid';
import * as simpleRepeat from './pages/blocks/simple-repeat.json';
import * as interpolation from './pages/blocks/interpolation.json';
import * as conditionalAttr from './pages/blocks/conditionalAttr.json';
import * as simpleConditional from './pages/blocks/simple-conditional.json';
import * as unlessLoop from './pages/blocks/unless-loop.json';
import * as inputWithChecked from './pages/blocks/input-with-checked.json';

import { format } from 'prettier';

test('Simple repeat', async () => {
  const output = format(blockToLiquid(simpleRepeat as any), {
    parser: 'html',
  });
  expect(output).toMatchSnapshot();
});

test('Variable interpolation', async () => {
  const output = format(blockToLiquid(interpolation as any), {
    parser: 'html',
  });
  expect(output).toMatchSnapshot();
});

test('Conditional Attributes', async () => {
  const output = format(blockToLiquid(conditionalAttr as any), {
    parser: 'html',
  });
  expect(output).toMatchSnapshot();
});

test('Handles simple conditional', async () => {
  const output = format(blockToLiquid(simpleConditional as any), {
    parser: 'html',
  });
  expect(output).toMatchSnapshot();
});

test('Handles loop with unless statement', async () => {
  // Prettier has trouble with liquid tags and does not work
  // to format the output from this test
  const output = blockToLiquid(unlessLoop as any);
  expect(output).toMatchSnapshot();
});

test('Handles input with boolean attribute checked', async () => {
  const output = format(blockToLiquid(inputWithChecked as any), {
    parser: 'html',
  });
  expect(output).toMatchSnapshot();
});
