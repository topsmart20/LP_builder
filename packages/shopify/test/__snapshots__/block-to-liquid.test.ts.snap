// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditional Attributes 1`] = `
"<style>
  .builder-block.builder-oxvj38205q {
  }
</style>

<div
  data-test=\\"{% if collection.image %} {{collection.image | img_url: sizes.first}}{% else  %} {% if section.settings.image %} {{section.settings.image | img_url: sizes.first}}{% else  %} {{collection.products.first.featured_image | product_img_url: sizes.first}}{% endif %}{% endif %}\\"
  alt=\\"{% if positive %} alt-positive{% endif %}\\"
  class=\\"builder-block builder-oxvj38205q {% if a %} a{% else  %} {% if b %} b{% else  %} willBeLost{% endif %}{% endif %}\\"
  builder-id=\\"builder-oxvj38205q\\"
></div>
"
`;

exports[`Handles input with boolean attribute checked 1`] = `
"<style>
  .builder-block.builder-oxvj38205q {
  }
</style>

<input
  type=\\"radio\\"
  name=\\"option-{{option_index}}\\"
  value=\\"{{value | escape}}\\"
  {%
  if
  forloop.first
  %}
  checked{%
  endif
  %}
  builder-id=\\"builder-oxvj38205q\\"
  class=\\"builder-block builder-oxvj38205q\\"
/>
"
`;

exports[`Handles loop with unless statement 1`] = `
"
    <style>.builder-block.builder-fgyfi20qi84 {}</style>
    
    
    
    
      <select  name=\\"tag_filter\\" id=\\"tag_filter\\" class=\\"builder-block builder-fgyfi20qi84 tag_filter\\" data-default-collection=\\"{% if collection.handle == 'all' %} /collections/all {% else %}{{collection.url}}{% endif %}\\" builder-id=\\"builder-fgyfi20qi84\\">
        
    <style>.builder-block.builder-prvgde6tkhl {}</style>
    {% for tag in collection.all_tags %}
    
    
    
    <style>.builder-block.builder-phovceulmn {}</style>
    
    
    
    {% unless tag contains 'meta-' %}
    <style>.builder-block.builder-51xxmnv9mj6 {}</style>
    
    
    
    
      <option  {% if current_tags contains tag %} selected{% endif %} value=\\"/collections/{% if collection.handle != blank %}{{collection.handle}}{% else %}all{% endif %}/{{tag | handleize}}\\" builder-id=\\"builder-51xxmnv9mj6\\" class=\\"builder-block builder-51xxmnv9mj6\\">
        
    <style>.builder-block.builder-qj4kte3xlb {
  display: inline;}</style>
    
    
    
    
      <span  builder-id=\\"builder-qj4kte3xlb\\" class=\\"builder-block builder-qj4kte3xlb\\">
        
      <style>
        .builder-text p:first-of-type, .builder-paragraph:first-of-type { margin: 0 } .builder-text > p, .builder-paragraph { color: inherit; line-height: inherit; letter-spacing: inherit; font-weight: inherit; font-size: inherit; text-align: inherit; font-family: inherit; }
      </style>
      <span class=\\"builder-text\\">
        {{ tag }}
      </span>
    
      </span>
    
    
    
  
      </option>
    
    
    
  {% else %}
    <style>.builder-block.builder-6f0s06v6h0j {
  display: inline;}</style>
    
    
    
    
      <span  builder-id=\\"builder-6f0s06v6h0j\\" class=\\"builder-block builder-6f0s06v6h0j\\">
        
      <style>
        .builder-text p:first-of-type, .builder-paragraph:first-of-type { margin: 0 } .builder-text > p, .builder-paragraph { color: inherit; line-height: inherit; letter-spacing: inherit; font-weight: inherit; font-size: inherit; text-align: inherit; font-family: inherit; }
      </style>
      <span class=\\"builder-text\\">
         
 you are  
      </span>
    
      </span>
    
    
    
  

    <style>.builder-block.builder-qqtgpdhbkg {
  display: inline;}</style>
    
    
    
    
      <span  builder-id=\\"builder-qqtgpdhbkg\\" class=\\"builder-block builder-qqtgpdhbkg\\">
        
      <style>
        .builder-text p:first-of-type, .builder-paragraph:first-of-type { margin: 0 } .builder-text > p, .builder-paragraph { color: inherit; line-height: inherit; letter-spacing: inherit; font-weight: inherit; font-size: inherit; text-align: inherit; font-family: inherit; }
      </style>
      <span class=\\"builder-text\\">
        {{ cool }}
      </span>
    
      </span>
    
    
    
  {% endunless %}
    
    
    
  
    
    
    {% endfor %}
  
      </select>
    
    
    
  "
`;

exports[`Handles simple conditional 1`] = `
"<style>
  .builder-block.builder-rcpqlcqsh3f {
  }
</style>

{% if collection.description != blank %}
<style>
  .builder-block.builder-4o01ryi0ztu {
  }
</style>

<div
  class=\\"builder-block builder-4o01ryi0ztu section clearfix collection_description\\"
  builder-id=\\"builder-4o01ryi0ztu\\"
>
  <style>
    .builder-block.builder-7jz57t7fk8a {
      display: inline;
    }
  </style>

  <span
    builder-id=\\"builder-7jz57t7fk8a\\"
    class=\\"builder-block builder-7jz57t7fk8a\\"
  >
    <style>
      .builder-text p:first-of-type,
      .builder-paragraph:first-of-type {
        margin: 0;
      }
      .builder-text > p,
      .builder-paragraph {
        color: inherit;
        line-height: inherit;
        letter-spacing: inherit;
        font-weight: inherit;
        font-size: inherit;
        text-align: inherit;
        font-family: inherit;
      }
    </style>
    <span class=\\"builder-text\\">
      {{ collection.description }}
    </span>
  </span>
</div>

{% elsif collection.title != blank %}
<style>
  .builder-block.builder-4o01ryi0ztu {
  }
</style>

<div
  class=\\"builder-block builder-4o01ryi0ztu section clearfix collection_title\\"
  builder-id=\\"builder-4o01ryi0ztu\\"
>
  <style>
    .builder-block.builder-7jz57t7fk8a {
      display: inline;
    }
  </style>

  <span
    builder-id=\\"builder-7jz57t7fk8a\\"
    class=\\"builder-block builder-7jz57t7fk8a\\"
  >
    <style>
      .builder-text p:first-of-type,
      .builder-paragraph:first-of-type {
        margin: 0;
      }
      .builder-text > p,
      .builder-paragraph {
        color: inherit;
        line-height: inherit;
        letter-spacing: inherit;
        font-weight: inherit;
        font-size: inherit;
        text-align: inherit;
        font-family: inherit;
      }
    </style>
    <span class=\\"builder-text\\">
      {{ collection.title }}
    </span>
  </span>
</div>

{% else %}
<style>
  .builder-block.builder-4o01ryi0ztu {
  }
</style>

<div
  class=\\"builder-block builder-4o01ryi0ztu section clearfix collection_title\\"
  builder-id=\\"builder-4o01ryi0ztu\\"
>
  <style>
    .builder-block.builder-7jz57t7fk8a {
      display: inline;
    }
  </style>

  <span
    builder-id=\\"builder-7jz57t7fk8a\\"
    class=\\"builder-block builder-7jz57t7fk8a\\"
  >
    <style>
      .builder-text p:first-of-type,
      .builder-paragraph:first-of-type {
        margin: 0;
      }
      .builder-text > p,
      .builder-paragraph {
        color: inherit;
        line-height: inherit;
        letter-spacing: inherit;
        font-weight: inherit;
        font-size: inherit;
        text-align: inherit;
        font-family: inherit;
      }
    </style>
    <span class=\\"builder-text\\">
      {{ collection.name }}
    </span>
  </span>
</div>

{% endif %}
"
`;

exports[`Simple repeat 1`] = `
"<style>
  .builder-block.builder-85yaddnm61v {
  }
</style>
{% for link in linklists[link.handle].links %}

<style>
  .builder-block.builder-66af795225934811ab67c5d5b2e81987 {
    display: flex;
    flex-direction: column;
  }
</style>

<div
  builder-id=\\"builder-66af795225934811ab67c5d5b2e81987\\"
  class=\\"builder-block builder-66af795225934811ab67c5d5b2e81987\\"
>
  <style>
    .builder-text p:first-of-type,
    .builder-paragraph:first-of-type {
      margin: 0;
    }
    .builder-text > p,
    .builder-paragraph {
      color: inherit;
      line-height: inherit;
      letter-spacing: inherit;
      font-weight: inherit;
      font-size: inherit;
      text-align: inherit;
      font-family: inherit;
    }
  </style>
  <span class=\\"builder-text\\">
    {{ link }}
  </span>
</div>

{% endfor %}
"
`;

exports[`Variable interpolation 1`] = `
"<style>
  .builder-block.builder-oxvj38205q {
  }
</style>

<a
  rel=\\"me\\"
  target=\\"_blank\\"
  class=\\"builder-block builder-oxvj38205q icon-youtube\\"
  href=\\"{{settings.youtube_link}}\\"
  title=\\"{{shop.name}} on YouTube\\"
  builder-id=\\"builder-oxvj38205q\\"
>
</a>
"
`;
